packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -40,113 +40,184 @@ DataGridRaw.defaultProps = DEFAULT_GRID_PROPS_FROM_OPTIONS;
 
 export const DataGrid = React.memo(DataGridRaw);
 
-// @ts-ignore
-DataGrid.propTypes = {
-  apiRef: chainPropTypes(PropTypes.any, (props: any) => {
-    if (props.apiRef != null) {
-      return new Error(
-        [
-          `Material-UI: \`apiRef\` is not a valid prop.`,
-          'GridApiRef is not available in the MIT version.',
-          '',
-          'You need to upgrade to the DataGridPro component to unlock this feature.',
-        ].join('\n'),
-      );
-    }
-    return null;
-  }),
-  checkboxSelectionVisibleOnly: chainPropTypes(PropTypes.bool, (props: any) => {
```

Comment:
We are losing the custom prop-types. If necessary, we can use this comment https://github.com/mui-org/material-ui/blob/a4e2ed82c76fffb5875ba96ab8c0efd5bcb424ef/packages/material-ui/src/Popper/Popper.js#L314 

x the others

Reply:
The comment doesn't work in this situation. I'm already using it on other props. The prop-type for this prop was removed because `checkboxSelectionVisibleOnly` is omitted from the grid's props. The generator doesn't allow to have more prop-types than the props specified in the TS definition.

https://github.com/mui-org/material-ui-x/blob/1a0ad36f7380ba6bff04fa8f2b6366e001988734/packages/grid/data-grid/src/DataGridProps.ts#L11

Reply:
What is the conclusion here ?
Do we have a way to improve the generator to keep those prop types ?

Reply:
There's no way to include more props than those allowed in the interface of the component. In theory, if the prop is not in the interface it shouldn't be used in the component. We could add a warning in `useDataGridProps` if the user tries to use a prop from the Pro version.

Reply:
Would you do this switch to `useDataGridProps` / `useDataGridProProps` in this PR ?
It would be sad to have a release without those warnings.

Reply:
> Would you do this switch to useDataGridProps / useDataGridProProps in this PR ?

In another PR.